Update to latest upstream

mratsim opened the pull request.

lynncrees committed:
Add default argument to 'tables.getOrDefault'

Commit:
Fix background-color in nimdoc.cfg

Don't assume, that the default background-color is white.
My default background-color is dark, which makes the documentation hardly readable: https://i.imgur.com/xN0UjWz.png
The reason is the transparency of the existing color (rgba(252, 248, 244, 0.45);) 
Fixed by removing the transparency.

yglukhov committed:
Convertion procs for (IpAddress, Port) <-> (SockAddr, Socklen)

yglukhov committed:
Fixed crash/interface. Added tests.

yglukhov committed:
More flexible fromSockAddr interface

jangko committed:
fixes #7332 (#7341)

Araq committed:
json.nim: documentation copy&paste issue

Araq committed:
Merge branch 'devel' of github.com:nim-lang/Nim into devel

Araq committed:
make 'memset' calls more robust; refs #7341

Araq committed:
Merge branch 'devel' of github.com:nim-lang/Nim into devel

Wh1teDuke committed:
[Documentation] free -> dealloc (#7334)

Araq committed:
make 'memset' calls more robust; refs #7341

PMunch committed:
Add CommentStmt to astGenRepr (#7313)

* Added codeRepr and dumpCode to the macros module.

This allows those writing macros to write examples, get the code to generate the AST for that example, and then modify that code to be dynamic with the macro function.

cooldome committed:
Codegen: use type forward declarations more aggresively. Fixes #7339 (#7340)

Do not emit object definition it if used only by ref or ptr

yglukhov committed:
More flexible lowerBound interface (#7344)

jangko committed:
fixes #7347, asyncfile.getFileSize (#7354)

* fixes #7347

* fixes #7347

twetzel59 committed:
Fix #7304 by clarifying integer width in manual (#7319)

* Fix #7304 by clarifying integer width in manual

* Update lexing.txt

jcosborn committed:
fix allocator corruption for large sizes (#7338)

* fix allocator corruption for large sizes
* allow large chunks to coalesce and added test case
* use correct constants in MaxBigChunkSize

Araq committed:
raise the max instruction count for the VM to one billion

GULPF committed:
Add string overload for parseXml

Araq committed:
better error messages

Araq committed:
fixes #7335

dom96 committed:
Merge pull request #7356 from GULPF/parseXml-string

Add string overload for parseXml

dom96 committed:
Merge pull request #7326 from trustable-code/patch-6

Fix background-color in nimdoc.cfg

Araq committed:
fixes #7331

dom96 committed:
Remove bugfixes heading from changelog

This heading is for the automatically generated list of issues fixed.

GULPF committed:
Remove basic2d & 3d from module list

`basic2d` and `basic3d` has been moved to nimble packages

Araq committed:
fixes #7364

Araq committed:
disable flaky fragmenation test for AppVeyor

Araq committed:
better error messages for illegal captures

dom96 committed:
Merge pull request #7368 from GULPF/patch-2

Remove basic2d & 3d from module list

cooldome committed:
Fixes #7363 (#7371)

SolitudeSF committed:
libssl.so.44 fix (#7358)

euantorano committed:
Add an example to the `dynlib` module doc

Following a quesiton on Gitter about loading libraries dynamically based upon some runtime parameter, I noticed there's no documentation on the `dynlib` module for something that's quite common.

dom96 committed:
Merge pull request #7388 from euantorano/patch-1

Add an example to the `dynlib` module doc

Araq committed:
further planning of Nim is done via RFCs; no todo.txt anymore

Araq committed:
nimsuggest: do not execute the .nims file when requested to auto-complete it

cooldome committed:
fixes #7392 (#7395)

Araq committed:
mysql.nim: fixes how the DLL version loading works

Araq committed:
more checking for 'var T' as return type; refs #7373

Araq committed:
enforce 'var T' produces a view into the first parameter; refs #7373

Araq committed:
fixes the 'var T' checking

Araq committed:
make tests green again

zah committed:
Static[T] fixes (#7333)

* fix the usage of unresolved static[T] parameters in proc signatures
* fix tsametype and tmacrogenerics
* Allow creating composite type classes with concepts and using them in type signatures
* Allow integers to be used in ident concatenations
* Support using imported C++ generic types in proc signatures
* fixes #7230
* closes #7379
* re-enable some metatype tests

Araq committed:
added toOpenArray builtin for zero-copy slices; syntax sugar yet to come

yglukhov committed:
Comments addressed

cabhishek committed:
fix small typo in the documentation (#7411)

timotheecour committed:
doc: fix path to $HOME/.config/nim.cfg to work on POSIX (not just unix but also osx) (#7421)

alehander92 committed:
Disable param filtering

gmfawcett committed:
net.connect (with timeout), raise error on connect failure

Under Linux (probably POSIX), the current code tests for timeout, but
does not test for connection failure. connectAsync() returns succesfully
upon an EINPROGRESS result; but at this point, the connection state is
still unknown. After selectWrite() is done, we need to test the socket
for errors again.

cooldome committed:
fixes #7414 (#7434)

dom96 committed:
Merge pull request #7436 from gmfawcett/fawcett-net-connect-error

net.connect (with timeout), raise error on connect failure

dom96 committed:
Merge pull request #7423 from alehander42/feature-disable-param-filtering

Unittest param filtering

dom96 committed:
Merge pull request #7075 from lcrees/get_default

Add default argument to tables.getOrDefault

dom96 committed:
Fixes #7433.

treeform committed:
Add some missing things to dom.nim

Add window.performance, window.performance.now(), window.devicePixelRatio, window.scrollX and Y, style.overflowX and Y, node.getBoundingRect.

dom96 committed:
Merge pull request #7437 from treeform/patch-2

Add some missing things to dom.nim

Araq committed:
further steps in implementing sink parameters; refs #7041

Araq committed:
Merge branch 'devel' of github.com:nim-lang/Nim into devel

andreaferretti committed:
Update website.ini

andreaferretti committed:
Added asyncfutures doc

dom96 committed:
Fixes #7400

dom96 committed:
Merge pull request #7445 from nim-lang/asyncstreams-doc

Added documentation for asyncstreams

cabhishek committed:
Fixes #6902

Araq committed:
destructors: first version of 'sink' parameter logic

data-man committed:
Comparing datetimes

dom96 committed:
Improve compilation instructions in readme

dom96 committed:
Some more fixes to the readme.

arnetheduck committed:
RLIMIT_NOFILE as posix const

Araq committed:
sink parameters: next steps

Araq committed:
Merge branch 'devel' of github.com:nim-lang/Nim into devel

dom96 committed:
Merge pull request #7456 from data-man/cmp_datetime

Comparing datetimes

dom96 committed:
Merge pull request #7336 from yglukhov/ipaddr-sockaddr-conversions

Conversion procs for (IpAddress, Port) <-> (SockAddr, Socklen)

dom96 committed:
Merge pull request #7459 from arnetheduck/posix-rlimit-nofile

RLIMIT_NOFILE as posix const

Araq committed:
fixes codegen for nil cstrings

Araq committed:
sysstr.nim: fix minor typos

yglukhov committed:
Fixed compilation of times.nim with mingw 4.9.2 C++ 32bit

cheatfate committed:
Fix compiler stuck on waiting C/CPP backends. (#7472)

Araq committed:
C codegen: preparations for different seq and string implementations

Araq committed:
Merge branch 'devel' of github.com:nim-lang/Nim into devel

romovs committed:
Fix documentation link for set type (#7465)

GULPF committed:
Don't assume UTC in cookies.setCookie

Araq committed:
implemented v2 string literals

krux02 committed:
fewer temporary string allocations for the compiler's renderer (#7484)

GULPF committed:
Move algorithm tests away from koch

GULPF committed:
Fix algorithm.fill for empty input

dom96 committed:
Merge pull request #7495 from GULPF/fix-fill-bug

Fix algorithm.fill for empty input

krux02 committed:
Fixes #7473 (#7475)

* removed code duplication
* changelog entry, adopt tests for changes

dom96 committed:
Merge pull request #7450 from cabhishek/devel

Fixes #6902

dom96 committed:
Merge pull request #7483 from GULPF/cookies-timezones

Don't assume UTC in cookies.setCookie

timotheecour committed:
cmd line help: clarify compileOnly vs noLinking

dom96 committed:
Merge pull request #7511 from timotheecour/pr_clarify_compileOnly

[trivial] [doc] cmd line help: clarify compileOnly vs noLinking

dom96 committed:
Merge pull request #7471 from yglukhov/fixed-times-cpp-windows-32

Fixed compilation of times.nim with mingw 4.9.2 C++ 32bit

skilchen committed:
include help short option in processCmdLine (#7515)

arnetheduck committed:
__NR_gettid as const on amd64 (#7460)

data-man committed:
Fixes #5281 (#7489)

skilchen committed:
fix issues #6986 and #7501 (#7513)

cooldome committed:
Fixes #7407 (#7427)

Araq committed:
the 'deprecated' pragma for modules now supports an error message

Araq committed:
make parseopt2 use the new better deprecated error message

Araq committed:
renamed future.nim to sugar.nim

Araq committed:
fixes a regression about static object case variant checking

Araq committed:
fixes #6843

Araq committed:
fixes #7522

Araq committed:
improve compiler error message

Araq committed:
revert the error message

GULPF committed:
semcheck negative array length (#7518)

Araq committed:
added the 'x.p[:T]' notation for explicit generic instantiations in combination with the ddot calling syntax

Araq committed:
fixes #7252

Araq committed:
fixes #7528

GULPF committed:
Fix parser bug with type classes (#7480)

Araq committed:
make tests green again

FedericoCeratto committed:
Add RISC-V (riscv64) support (#7417)

FedericoCeratto committed:
Allow searching for "rename file" in documentation

dom96 committed:
Merge pull request #7536 from FedericoCeratto/patch-3

Allow searching for "rename file" in documentation

dom96 committed:
Call requestAux not request in httpclient.

GULPF committed:
Fix compile time set cardinality (#7558)

cooldome committed:
Cpp codegen: handling of imported exceptions. Fixes #3571 (#7360)

genotrance committed:
Add a few useful os calls to nimscript (#7442)

Araq committed:
concepts: do not report concept parameters as unused

Araq committed:
fixes #7551

FedericoCeratto committed:
Enable RISC-V 64bit platform (#7541)

Related to  #7417

krux02 committed:
Get symbol kind (#7491)

krux02 committed:
option optimization (#6253)

zah commented:
I think we'll be better off rebasing against devel instead of merging it. This will lead to a little bit of inconvenience from time to time (you'll have to use force pushes and fetches when migrating to a new version), but the benefit is that our patches will be much easier to review (they will always appear as the last few commits following a certain devel HEAD).

Right now in particular, our patches got squash-merged in the Nim repo, so we can just force-push Nim's devel over the nimbus branch.

Araq committed:
fixes #6895

Araq committed:
fixes #2779

cooldome committed:
quick bug fix (#7576)

krux02 committed:
fix deprecation warnings in asyncmacro and htmlgen (#7573)

* fix deprecation warnings in asyncmacro and htmlgen

* removed todo

genotrance committed:
Added a few useful os calls to VM (#7440)

arnetheduck committed:
Cleanups 20180401 (#7458)

aguspiza committed:
Move RLimit and getrlimit to posix.nim and add setrlimit to easily limit FD allocation (#7564)

treeform committed:
Added count(*) support to sql parser. Fixed warnings in sql parser. (#7490)

jxy committed:
unittest: default no color if stdout is not a tty (#7424)

We accept a new environment variable, NIMTEST_COLOR,
which override the effect of NIMTEST_NO_COLOR.
The environment variable, NIMTEST_COLOR, can be 'never'
or 'always', which set the color output to false or true,
respectively.

PMunch committed:
Add column number to instantiation info (#7376)

* Add column number to instantiation info

Instantiation info left out column number for no good reason. This adds
it in as the third element of the tuple.

* Fix test that failed and added change to changelog

An assertion test failed because it was declaring a type that was
expected to be populated by instantiationInfo with the old signature.

Also added the changes to the changelog as it is a breaking change.

zah commented:
Furthermore, our branch is actually called `nimbus`. Nobody should be using the devel branch here (which this pull request is trying to update).
